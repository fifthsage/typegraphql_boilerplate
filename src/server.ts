import express from "express";
import session from "express-session";
import apollo from "./apollo";
import logger from "morgan";
import helmet from "helmet";
import cors from "cors";
import common from "../config/common";
import { authenticateJwt } from "./passport";
import databaseConn from "./helpers/databaseConn";

const app = express();

app.use(
  session({
    secret: "keyboard cat",
    resave: false,
    saveUninitialized: false,
    cookie: { secure: false }
  })
);
app.use(
  cors({
    origin: common.corsDomain,
    optionsSuccessStatus: 200
  })
);

app.use(logger("dev"));
app.use(helmet());
app.use(authenticateJwt);

const prepareServer = async () => {
  try {
    const apolloServer = await apollo();

    apolloServer.applyMiddleware({
      app,
      path: "/api"
    });

    await databaseConn();
  } catch (e) {
    console.log(e);
  }

  return app;
};

const listen = () => {
  app.listen(common.port, () => {
    console.log(`ğŸš€ Server ready at ${common.port}`);
  });
};

export default async () => ({
  app: await prepareServer(),
  listen: listen
});
