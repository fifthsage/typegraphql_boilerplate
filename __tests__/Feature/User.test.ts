import request from "supertest";
import server from "../../src/server";

let app: any, apiServer: any;

beforeAll(async done => {
  const instance = await server();

  app = instance.app;

  apiServer = app.listen(done);
});

afterAll(async done => {
  await apiServer.close(done);
});

describe("User api", () => {
  it("users", async done => {
    const response = await request(app)
      .post("/api")
      .send({
        query: `query {
            users {
              id
              userName
              createdAt
              updatedAt
            }
          }`
      })
      .expect(200);

    expect(response.body).toMatchSnapshot();
    expect(Object.keys(response.body)).toStrictEqual(["data"]);
    expect(Object.keys(response.body.data)).toStrictEqual(["users"]);
    done();
  });
});
