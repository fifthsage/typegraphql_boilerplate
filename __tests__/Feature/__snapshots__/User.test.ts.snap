// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User api users 1`] = `
Object {
  "data": Object {
    "users": Array [],
  },
}
`;
